chain/messagesigner/messagesigner.go

```diff
@@ -66,15 +69,24 @@ func (ms *MessageSigner) SignMessage(ctx context.Context, msg *types.Message, sp
 	// Sign the message with the nonce
 	msg.Nonce = nonce
 
+	keyInfo, err := ms.wallet.WalletExport(ctx, msg.From)
+	if err != nil {
+		return nil, err
+	}
+	sb, err := SigningBytes(msg, key.ActSigType(keyInfo.Type))
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Did you want to add `KeyType` to the wallet api interface so you could call it here? This seems to do the same sort of operation.

Reply:
Removed KeyType